lib/resty/core/regex.lua

```diff
@@ -479,6 +480,71 @@ function ngx.re.find(subj, regex, opts, ctx, nth)
 end
 
 
+local function iterate_re_gmatch(self)
+    local compile_once = self._compile_once
+    local compiled = self._compiled
+    local subj = self._subj
+    local flags = self._flags
+    local pos = self._pos
+    -- The iterator is exhausted.
+    if not pos then
+        return nil
+    end
+
+    local rc = C.ngx_http_lua_ffi_exec_regex(compiled, flags, subj, #subj, pos)
```

Comment:
Maybe we could cache the subject length in `self._subj_len` to avoid calling the length operator on each iteration?